.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -9,3 +9,7 @@ TODO: Justify why this contribution should be part of the project.
 ### What testing has been done?
 
 TODO: Detail what testing has been done to ensure this submission meets requirements.
+
+### Signoff
+
+[ ] The authors agree to the Developer Certificate of Origin in `CONTRIBUTING.md`.
```

Comment:
This seems inferior to enforcing people sign off commits using `Signed-off-by`. GitHub has apps to check that, like [this one](https://github.com/apps/dco).

Rationale: If we provide a checkbox here, it's something we need to manually/visually inspect. Sure, it's not hard. But it's a single-bit information, easier to miss than the DCO app's red checkmark.

If GitHub were to go away, or we lose access to the pull request, then we wouldn't have a written record of DCO approval by committers. If we enforce a signed-off-by, then it's in the repository.

And finally, it's making it a bit too easy for people to make PRs compliant. This is something where the little bit of additional friction is actually preferable to to potential legal issues later on, even if it's as "simple" as having to revert people's code at a later point in time.

Reply:
@mbr0wn @alejandro5042 should we remove this line from [Requirements for open source projects](https://github.com/ni/.github/blob/main/profile/CONTRIBUTING.md)?

“Include language in .github/pull_request_template.md that explicitly acknowledges the submission adheres to the DCO”

Reply:
> This seems inferior to enforcing people sign off commits using `Signed-off-by`.

The [repo template](https://github.com/ni/github-repo-template/blob/8359f26efec5d2bd11d598add3c9b7caa7237110/CONTRIBUTING.md?plain=1#L12C1-L12C1) has this stronger requirement in its `CONTRIBUTING.md`:

> Please remember to sign off your commits (e.g., by using `git commit -s` if you
> are using the command line client). This amends your git commit message with a line
> of the form `Signed-off-by: Name Lastname <name.lastmail@emailaddress.com>`.

So I don't see how this a replacement or downgrade, but instead a both-and that meets the requirement that @benmont pointed out.

I browsed some pages in the repo settings, but I only found a toggle for _"Require contributors to sign off on web-based commits,"_ and none that required it for all commits. If there is a setting to highlight in [.github/profile/CONTRIBUTING.md](https://github.com/ni/.github/blob/main/profile/CONTRIBUTING.md#initial-setup), I can open a PR for that too.

Reply:
Wait, those lines seem familiar...

![image](https://github.com/ni/github-repo-template/assets/508035/12cb0026-6f0d-4937-94ed-99bca8a81078)

Oh right, I put them there :smile: 

I agree with you that this is the less strong requirement. But that also means this is a subset of what we already have.

Instead of this checkbox, I will suggest that all new projects enable the DCO sign-off app. You are correct, there is no way to enforce signoffs when not using the GitHub page (because of how git works independently from GitHub). But I do think that this checkbox carries the risk of someone not checking git commits manually (which they would have to do without such a DCO signoff app). However, with the app, this checkbox becomes completely redundant.

@benmont Yeah thanks for pointing out that file. I think we should change that line to something stronger, e.g., "use an automated tool (e.g. <INSERT LINK>) to verify that contributors are using signed-off-by lines").

Reply:
@jfriedri-ni @benmont What are your thoughts on using the DCO app? I could update the general CONTRIBUTING file.

Reply:
> What are your thoughts on using the DCO app? I could update the general CONTRIBUTING file.

Seems like a good idea :)

Reply:
I add this PR to the main repository: https://github.com/ni/.github/pull/1

@jfriedri-ni Can you please remove this section or replace it with something non-interactive (you must sign off commits... this will be checked automatically).

I'm OOO starting EOB tomorrow so I'll approve this and assume you can handle this. @benmont  I wasn't able to assign an owner for that PR mentioned above, can you merge it?

Reply:
Yes I can merge after @jfriedri-ni removes/replaces with something.